kv/kvserver/concurrency: TestLockTableConcurrentRequests failed

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/292d94d9-b983-485e-a6c5-10d8d080673f?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [911f1ce389342459592d89fe7b78bcba3a2d265a](https://github.com/cockroachdb/cockroach/commits/911f1ce389342459592d89fe7b78bcba3a2d265a):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_2 (24m57s)

```
Stack: 

```
goroutine 261024 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests4202612793
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_2
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37695

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/4e482329-1bb6-443c-844d-8b6b329257b6?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [f117eea22dd7be380c7141cdf6cd7aba92dd9c70](https://github.com/cockroachdb/cockroach/commits/f117eea22dd7be380c7141cdf6cd7aba92dd9c70):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_8 (24m54s)

```
Stack: 

```
goroutine 833581 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests2351147515
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_2
items: 1692, aborted: 0, concurrency violations: 758, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_8
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/bc37a5b6-7ec6-410d-b4f6-c4a68bb1982e?testReportRun=2&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [67547d7724f8a52646e2e8ecb3ca48b923957d90](https://github.com/cockroachdb/cockroach/commits/67547d7724f8a52646e2e8ecb3ca48b923957d90):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_8 (24m55s)

```
Stack: 

```
goroutine 1033150 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests3274046890
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_2
items: 1671, aborted: 0, concurrency violations: 661, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_8
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/97ab0b0b-e015-427e-9e00-9e90a6fbec88?testReportRun=2&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on release-24.1 @ [6e3e9e6012dc04dabc64333a87e77a80d9f9d46b](https://github.com/cockroachdb/cockroach/commits/6e3e9e6012dc04dabc64333a87e77a80d9f9d46b):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_16 (24m39s)

```
Stack: 

```
goroutine 2588996 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests1983242438
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_16
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/bc4cd6ca-86d1-40ea-9985-08fd871de137?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [8daa6933836c8ce1108fb22bf6663117fc71599b](https://github.com/cockroachdb/cockroach/commits/8daa6933836c8ce1108fb22bf6663117fc71599b):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_16 (24m46s)

```
Stack: 

```
goroutine 2996049 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests4126696952
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_2
items: 1677, aborted: 0, concurrency violations: 861, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_8
items: 1677, aborted: 0, concurrency violations: 857, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_16
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/818bb101-19a4-40be-abc7-466bb69d6978?testReportRun=2&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [737df7fe75c5698d9ba384ad322f30963582cc42](https://github.com/cockroachdb/cockroach/commits/737df7fe75c5698d9ba384ad322f30963582cc42):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_2 (24m57s)

```
Stack: 

```
goroutine 196958 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests3562098866
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_2
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/23a8157d-e05a-4a9e-a7a9-4c1578870909?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [6e94cb3b7992de4dec7688750cfd13971dd2930c](https://github.com/cockroachdb/cockroach/commits/6e94cb3b7992de4dec7688750cfd13971dd2930c):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_4 (9m32s)

```
Stack: 

```
goroutine 265982376 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests127189903
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1609: numRequests: 10000; numActiveTxns: 4; probability(txn-al reqs): 0.75; probability(new txns): 0.25; probability(duplicate access): 0.50
    lock_table_test.go:1734: txns creted: 1874
=== RUN   TestLockTableConcurrentRequests/concurrency_1
items: 11874, aborted: 0, concurrency violations: 11131, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_4
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/5162640e-6a93-4ca2-a5cb-26ba6e6d1bf1?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on release-24.1 @ [3111a51bdb2cbf889c92f4440bf88b0eb682cb96](https://github.com/cockroachdb/cockroach/commits/3111a51bdb2cbf889c92f4440bf88b0eb682cb96):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_16 (24m44s)

```
Stack: 

```
goroutine 2930059 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests2180491673
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestLockTableConcurrentRequests/concurrency_16
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/dbfbec68-cd3a-46d8-85bb-9f453a957fe6?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [e0068814dfcb4f975a53b79b5546b5fb85c0f927](https://github.com/cockroachdb/cockroach/commits/e0068814dfcb4f975a53b79b5546b5fb85c0f927):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_32 (21s)

```
Stack: 

```
goroutine 315646842 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests103809738
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1609: numRequests: 10000; numActiveTxns: 16; probability(txn-al reqs): 0.75; probability(new txns): 0.25; probability(duplicate access): 0.50
    lock_table_test.go:1734: txns creted: 1837
=== RUN   TestLockTableConcurrentRequests/concurrency_16
items: 11837, aborted: 0, concurrency violations: 11485, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_32
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/fca7632d-70b5-4f4a-95a4-ed3ba6b035ff?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [347cdc76d4c5abb2e872f325e944337a46b5883f](https://github.com/cockroachdb/cockroach/commits/347cdc76d4c5abb2e872f325e944337a46b5883f):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_8 (12m22s)

```
Stack: 

```
goroutine 350542742 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests4178933035
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1609: numRequests: 10000; numActiveTxns: 8; probability(txn-al reqs): 0.90; probability(new txns): 0.10; probability(duplicate access): 0.50
    lock_table_test.go:1734: txns creted: 917
=== RUN   TestLockTableConcurrentRequests/concurrency_2
items: 10917, aborted: 0, concurrency violations: 10537, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_8
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/1177f4db-c123-4d14-9156-1304a0afc3b4?testReportRun=2&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [e215cf435851cb9490fe4494fe73d02041c4d2c0](https://github.com/cockroachdb/cockroach/commits/e215cf435851cb9490fe4494fe73d02041c4d2c0):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_4 (12m9s)

```
Stack: 

```
goroutine 368018592 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests613030006
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1609: numRequests: 10000; numActiveTxns: 4; probability(txn-al reqs): 0.75; probability(new txns): 0.10; probability(duplicate access): 0.50
    lock_table_test.go:1734: txns creted: 753
=== RUN   TestLockTableConcurrentRequests/concurrency_1
items: 10753, aborted: 0, concurrency violations: 10204, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_4
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/b6d828bf-c58b-43c0-9cfe-3f395b56a878?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [0fbef53e48c2a3784631a09a42236e1d0636a5a7](https://github.com/cockroachdb/cockroach/commits/0fbef53e48c2a3784631a09a42236e1d0636a5a7):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_4 (11m8s)

```
Stack: 

```
goroutine 358365045 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests2386894273
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1609: numRequests: 10000; numActiveTxns: 2; probability(txn-al reqs): 0.75; probability(new txns): 0.25; probability(duplicate access): 0.50
    lock_table_test.go:1734: txns creted: 1888
=== RUN   TestLockTableConcurrentRequests/concurrency_2
items: 11888, aborted: 0, concurrency violations: 11115, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_4
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver/concurrency.TestLockTableConcurrentRequests [failed](https://tanzanite.cluster.engflow.com/invocations/default/118d65a0-d1bb-4274-9dba-17f97d0ab64f?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXIvY29uY3VycmVuY3k6Y29uY3VycmVuY3lfdGVzdA==) on master @ [fe6496ccabf2d4b2234cf8418004e5863847c0bd](https://github.com/cockroachdb/cockroach/commits/fe6496ccabf2d4b2234cf8418004e5863847c0bd):

Fatal error:

```
panic: test timed out after 24m57s
running tests:
	TestLockTableConcurrentRequests (24m57s)
	TestLockTableConcurrentRequests/concurrency_4 (10m55s)

```
Stack: 

```
goroutine 329327927 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestLockTableConcurrentRequests
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLockTableConcurrentRequests829779615
    test_log_scope.go:81: use -show-logs to present logs inline
    lock_table_test.go:1609: numRequests: 10000; numActiveTxns: 2; probability(txn-al reqs): 0.75; probability(new txns): 0.25; probability(duplicate access): 0.50
    lock_table_test.go:1734: txns creted: 1845
=== RUN   TestLockTableConcurrentRequests/concurrency_2
items: 11845, aborted: 0, concurrency violations: 10743, lock table: num=0

=== RUN   TestLockTableConcurrentRequests/concurrency_4
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLockTableConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/concurrency/lock_table_test.go
+++ b/pkg/kv/kvserver/concurrency/lock_table_test.go
@@ -1114,6 +1114,8 @@ func doWork(ctx context.Context, item *workItem, e *workloadExecutor) error {
 				g = nil
 			}
 		}()
+		var timer timeutil.Timer
+		defer timer.Stop()
 		var err error
 		for {
 			// Since we can't do a select involving latch acquisition and context
@@ -1137,10 +1139,18 @@ func doWork(ctx context.Context, item *workItem, e *workloadExecutor) error {
 			var lastID uuid.UUID
 		L:
 			for {
+				timer.Reset(time.Minute * 5)
 				select {
 				case <-g.NewStateChan():
 				case <-ctx.Done():
 					return ctx.Err()
+				case <-timer.C:
+					timer.Read = true
+					return errors.AssertionFailedf(
+						"request %d has been waiting for more than 5 minutes; lock table state:\n%s\n",
+						g.(*lockTableGuardImpl).seqNum,
+						e.lt.String(),
+					)
 				}
 				state, err := g.CurState()
 				if err != nil {
@@ -1582,6 +1592,13 @@ func TestLockTableConcurrentRequests(t *testing.T) {
 	probCreateNewTxn := possibleProbCreateNewTxn[rng.Intn(len(possibleProbTxnalReq))]
 	probDupAccessWithWeakerStr := 0.5
 	probOnlyRead := possibleProbOnlyRead[rng.Intn(len(possibleProbOnlyRead))]
+
+	if syncutil.DeadlockEnabled {
+		// We've seen 10,000 requests to be too much when running a deadlock build.
+		// Override numRequests to the lowest option (1,000) for deadlock builds.
+		numRequests = 1000
+	}
+
 	testLockTableConcurrentRequests(
 		t, numActiveTxns, numRequests, probTxnalReq, probCreateNewTxn,
 		probDupAccessWithWeakerStr, probOnlyRead,

